src/Board.cpp

```diff
@@ -0,0 +1,116 @@
+#include "Board.h"
+
+void Board::setBoard(const std::string& board) {
+	m_piece.resize(8); // resize the 2D vector
+	for (auto& row : m_piece) row.resize(8); // resize each row
+
+	// loop through the board string and create the corresponding piece objects
+	for (int i = 0; i < board.size(); ++i) {
+		int x = i % 8; // calculate x and y indices for the spot
+		int y = i / 8;
+		m_piece[y][x] = Factory<Piece>::create(board[i]);
+	}
+	setKingsPositions();
+}
+
+void Board::setKingsPositions() {
+	for (int i = 0; i < BoardSize; i++)
+		for (int j = 0; j < BoardSize; j++) 
+			if (m_piece[i][j] && typeid(*m_piece[i][j]) == typeid(King)) {
+				m_piece[i][j]->isWhite() ? setKingPos({ i + 1, j + 1 }, true) :
+					setKingPos({ i + 1, j + 1 }, false);
+			}
+}
+
+int Board::getResponse(const std::string& input) {
+	// parse the input string
+	Position src = { input[0] - 'a', input[1] - '1' };
+	Position dst = { input[2] - 'a', input[3] - '1' };
+
+	// board checks
+
+	// check if the src is empty
+	if (!m_piece[src.x][src.y]) return 11;
+
+	// check if the src is occupied by the opponent
+	if (m_piece[src.x][src.y]->isWhite() != m_whiteMove) return 12;
+
+	// check if the dst is occupied by the same color of the current player
+	if (m_piece[dst.x][dst.y] && m_piece[src.x][src.y]->isWhite() == m_piece[dst.x][dst.y]->isWhite())
+		return 13;
+
+	// check if the player move is legal
+	bool opponent = m_piece[dst.x][dst.y] && m_piece[src.x][src.y]->isWhite() == !m_piece[dst.x][dst.y]->isWhite();
+	if (!m_piece[src.x][src.y]->canMove(src, dst, opponent) || !checkIfNextStepClear(src, dst)) return 21;
+
+	// if player move was legal
+	m_piece[dst.x][dst.y] = move(m_piece[src.x][src.y]);
```

Comment:
This is dangerous. You're moving the `unique_ptr` from `src`, but not cleaning up. 
This could cause your code to crash (if in some next move someone tries to use the pointer in src)